### PR TITLE
feat: added floating badge functionality 

### PR DESCRIPTION
**Pull request by anuraghazra**

### CODE REVIEW

File: src/badge/Badge.tsx [hunk 1]
```diff
@@ -18,21 +18,67 @@ export type BadgeProps = BoxProps & {
...
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+      <Box
+        as="span"
+        ref={innerRef => {
+          ref = innerRef;
+          htmlref.current = innerRef;
+        }}
```
Comment (navin-moorthy) [inline src/badge/Badge.tsx:77]: May be use a custom hook for this 🤔

File: src/badge/Badge.tsx [hunk 2]
```diff
@@ -18,21 +18,67 @@ export type BadgeProps = BoxProps & {
...
+            !props.children ? theme.badge.dot[size] : "",
+          )
+        : "",
       className,
     );
 
-    return <Box as="span" ref={ref} className={badgeStyles} {...rest} />;
+    React.useEffect(() => {
+      if (!floating) return;
+      if (ref && htmlref.current) {
```
Comment (navin-moorthy) [inline src/badge/Badge.tsx:64]: I don't think its necessary to check both the ref 😑

File: src/badge/stories/Badge.stories.tsx [hunk 3]
```diff
@@ -2,17 +2,26 @@ import React from "react";
 import { Meta } from "@storybook/react/types-6-0";
 
 import {
-  createControls,
   storyTemplate,
+  createUnionControl,
 } from "../../../.storybook/storybookUtils";
 import { Badge, BadgeProps } from "../Badge";
+import { Button } from "../../button";
```
Comment (navin-moorthy) [inline src/badge/stories/Badge.stories.tsx:9]: Arrange the imports at last.

File: src/badge/Badge.tsx (hunk 2 above)
Comment (anuraghazra) [reply src/badge/Badge.tsx:64]: thats a mistake

**Pull request merged**